whirlpool/appliancesmanager.py

```diff
@@ -117,18 +143,72 @@ async def fetch_appliances(self):
 
         return success_owned or success_shared
 
-    @property
-    def aircons(self):
-        return self._aircons
-
-    @property
-    def refrigerators(self):
-        return self._refrigerators
+    async def fetch_all_data(self):
+        for appliance in self.all_appliances:
+            await appliance.fetch_data()
+
+    async def connect(self):
+        """Connect to appliance event listener"""
+        await self.start_event_listener()
+
+    async def disconnect(self):
+        """Disconnect from appliance event listener"""
+        await self.stop_event_listener()
+
+    async def start_event_listener(self):
+        """Start the appliance event listener"""
+        await self.fetch_all_data()
+        if self._event_socket is not None:
+            LOGGER.warning("Event socket not None when starting event listener")
+
+        self._event_socket = EventSocket(
+            await self._getWebsocketUrl(),
+            self._auth,
+            list({
+                **self._aircons,
+                **self._washerdryers,
+                **self._ovens,
+                **self._refrigerators,
+            }.keys()),
+            self._event_socket_callback,
+            self.fetch_all_data,
+            self._session,
+        )
+        self._event_socket.start()
+
+    async def stop_event_listener(self):
+        """Stop the appliance event listener"""
+        await self._event_socket.stop()
+        self._event_socket = None
+
+    def _event_socket_callback(self, msg: str):
+        json_msg = json.loads(msg)
+        said = json_msg["said"]
+        app = {
+            **self._aircons,
+            **self._washerdryers,
+            **self._ovens,
+            **self._refrigerators
+        }.get(said)
+        if app is None:
+            LOGGER.warning(f"Received message for unknown appliance {said}")
```

Comment:
```suggestion
            LOGGER.warning("Received message for unknown appliance %s", said)
```
see [docs.astral.sh/ruff/rules/logging-f-string#why-is-this-bad](https://docs.astral.sh/ruff/rules/logging-f-string/#why-is-this-bad) regarding the strings in logs (there are a lot more on the library that I hope to clean up eventually)